submodule.go

```diff
@@ -90,7 +93,7 @@ func (s *Submodule) status(idx *index.Index) (*SubmoduleStatus, error) {
 		err = nil
 	}
 
-	return status, err
+	return status, nil
```

Comment:
Are we intentionally discarding the error here? If so, the previous if can be removed.